platforms/core-execution/snapshots/src/main/java/org/gradle/internal/snapshot/impl/ImplementationSnapshot.java

```diff
@@ -35,7 +35,7 @@
  */
 public abstract class ImplementationSnapshot implements ValueSnapshot {
 
-    private static final String GENERATED_LAMBDA_CLASS_SUFFIX = "$$Lambda$";
+    private static final String GENERATED_LAMBDA_CLASS_SUFFIX = "$$Lambda";
```

Comment:
Is this Java version specific?

Reply:
Technically yes, but I don't think it really matters enough to enforce the slightly stricter check on pre-21.